### PR TITLE
Feat/taskitem progress

### PR DESCRIPTION
**Pull request by loicmathieu**
Send a worker task result for : 
- The start of running of a worker task execution
- The end of running of a worker task execution

Thanks to that, outputs are computed in real-time for iterations of executable tasks.

image

### CODE REVIEW

**Comment by loicmathieu** [comment]
A screenshot from the new UI part:

image

**Review by anna-geller — CHANGES_REQUESTED** [review]
great work! One thing we need to improve is that we should expand this view by default in the same way as other `ForEach` tasks expand dynamic tasks by default

image

**Comment by Skraye** [comment]
close #2135

File: core/src/main/java/io/kestra/core/tasks/flows/Flow.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
         )
     }
 )
-public class Flow extends Task implements ExecutableTask {
+public class Flow extends Task implements ExecutableTask<Flow.Output> {
```
Comment (tchiotludo) [inline core/src/main/java/io/kestra/core/tasks/flows/Flow.java:46]: The documentation is still not generated

File: core/src/main/java/io/kestra/core/tasks/flows/ForEachItem.java [hunk 2]
```diff
@@ -231,4 +229,14 @@ public static class Batch implements StorageSplitInterface {
...
     }
+
+    @Builder
+    @Getter
+    public static class Output implements io.kestra.core.models.tasks.Output {
+        @Schema(
+            title = "The iterations counter.",
+            description = "This output will be updated in real-time with the subflow executions.\n It will contains one counter by subflow execution state, plus a `max` counter that represent the maximum number of iterations (or the number of batches)."
+        )
+        private final Map<String, Integer> iterations;
```
Comment (tchiotludo) [inline core/src/main/java/io/kestra/core/tasks/flows/ForEachItem.java:240]: THe output is useless like this, please use strong typing: 
```java 
private Map<State.Type, Integer> states
private Integer max
...
```

**Review by tchiotludo — CHANGES_REQUESTED** [review]

**Pull request merged**